core/src/main/java/gov/nasa/jpl/mudrod/driver/ESDriver.java

```diff
@@ -245,135 +248,161 @@ public String searchByQuery(String index, String type, String query, Boolean bDe
     QueryBuilder qb = QueryBuilders.queryStringQuery(query);
     SearchResponse response = getClient().prepareSearch(index).setTypes(type).setQuery(qb).setSize(500).execute().actionGet();
 
-    Gson gson = new Gson();
-    List<JsonObject> fileList = new ArrayList<>();
+    // Map of K,V pairs where key is the field name from search result and value is the that should be returned for that field. Not always the same.
+    Map<String, String> fieldsToReturn = new HashMap<>();
+
+    fieldsToReturn.put("Dataset-ShortName", "Short Name");
+    fieldsToReturn.put("Dataset-LongName", "Long Name");
+    fieldsToReturn.put("DatasetParameter-Topic", "Topic");
+    fieldsToReturn.put("Dataset-Description", "Dataset-Description");
+    fieldsToReturn.put("DatasetCitation-ReleaseDateLong", "Release Date");
+
+    if (bDetail) {
+      fieldsToReturn.put("DatasetPolicy-DataFormat", "DataFormat");
+      fieldsToReturn.put("Dataset-Doi", "Dataset-Doi");
+      fieldsToReturn.put("Dataset-ProcessingLevel", "Processing Level");
+      fieldsToReturn.put("DatasetCitation-Version", "Version");
+      fieldsToReturn.put("DatasetSource-Sensor-ShortName", "DatasetSource-Sensor-ShortName");
+      fieldsToReturn.put("DatasetProject-Project-ShortName", "DatasetProject-Project-ShortName");
+      fieldsToReturn.put("DatasetParameter-Category", "DatasetParameter-Category");
+      fieldsToReturn.put("DatasetLocationPolicy-BasePath", "DatasetLocationPolicy-BasePath");
+      fieldsToReturn.put("DatasetParameter-Variable-Full", "DatasetParameter-Variable-Full");
+      fieldsToReturn.put("DatasetParameter-Term-Full", "DatasetParameter-Term-Full");
+      fieldsToReturn.put("DatasetParameter-VariableDetail", "DatasetParameter-VariableDetail");
+
+      fieldsToReturn.put("DatasetRegion-Region", "Region");
+      fieldsToReturn.put("DatasetCoverage-NorthLat", "NorthLat");
+      fieldsToReturn.put("DatasetCoverage-SouthLat", "SouthLat");
+      fieldsToReturn.put("DatasetCoverage-WestLon", "WestLon");
+      fieldsToReturn.put("DatasetCoverage-EastLon", "EastLon");
+      fieldsToReturn.put("DatasetCoverage-StartTimeLong-Long", "DatasetCoverage-StartTimeLong-Long");
+      fieldsToReturn.put("Dataset-DatasetCoverage-StopTimeLong", "Dataset-DatasetCoverage-StopTimeLong");
+
+      fieldsToReturn.put("Dataset-TemporalResolution", "Dataset-TemporalResolution");
+      fieldsToReturn.put("Dataset-TemporalRepeat", "Dataset-TemporalRepeat");
+      fieldsToReturn.put("Dataset-LatitudeResolution", "Dataset-LatitudeResolution");
+      fieldsToReturn.put("Dataset-LongitudeResolution", "Dataset-LongitudeResolution");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Dataset-LongitudeResolution" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)